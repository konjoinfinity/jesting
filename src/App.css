.App {
  text-align: center;
}

.App-logo {
  height: 20vmin;
  pointer-events: none;
}

.header-pink {
  background-color: pink;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.header-purple {
  background-color: purple;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

h1 {
  font-size: 24px;
  margin: 10px 0 0 0;
  font-weight: lighter;
  text-transform: uppercase;
  color: #eeeeee;
}

p {
  font-size: 12px;
  font-weight: light;
  color: #333333;
}

span a {
  font-size: 18px;
  color: #cccccc;
  text-decoration: none;
  margin: 0 10px;
  transition: all 0.4s ease-in-out;
  &:hover {
    color: #ffffff;
  }
}

/* @keyframes float {
  0% {
    box-shadow: 0 5px 15px 0px rgba(0,0,0,0.6);
    transform: translatey(0px);
  }
  50% {
    box-shadow: 0 25px 15px 0px rgba(0,0,0,0.2);
    transform: translatey(-20px);
  }
  100% {
    box-shadow: 0 5px 15px 0px rgba(0,0,0,0.6);
    transform: translatey(0px);
  }
} */

.container {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.avatar { 
  /* height:180;
  width: 180; */
  border-radius: 50%;
  /* overflow: hidden;
  box-shadow: 0 5px 15px 0px rgba(0,0,0,0.6);
  transform: translatey(0px); */
 animation: growShrink 6s ease-in-out infinite;
}

@keyframes growShrink {
  0% {
      transform: scale(1);
  }
  50% {
      transform: scale(3);
  }
  100% {
    transform: scale(1);
  }
}

.content {
  width: 100%;
  max-width: 600px;
  padding: 20px 40px;
  box-sizing: border-box;
  text-align: center;
}

.suppoprt-me {
  display: inline-block;
  position: fixed;
  bottom: 10px;
  left: 10px;
  width: 20vw;
  max-width: 250px;
  min-width: 200px;
  z-index: 9;
  img {
    width: 100%;
    height: auto;
  }
}
