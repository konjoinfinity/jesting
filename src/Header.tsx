import * as React from 'react'

interface Header {
    color?: "pink" | "purple"
    children: React.ReactNode
}

const Header: React.FC<Header> = ({ color, children }) => {
  return <header className={`header header-${color}`}>{children}</header>
  }

export { Header }