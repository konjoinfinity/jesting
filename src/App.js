import logo from './logo.svg';
import './App.css';
import { Header } from './Header.tsx';

function App() {
  return (
    <div className="App">
      <Header title="header" className="testing" color="purple">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Konjo
        </p>
        </Header>
    </div>
  );
}

export default App;
