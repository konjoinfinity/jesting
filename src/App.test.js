import { render, screen } from '@testing-library/react';
import App from './App';
import {Header} from './Header'

test('renders Konjo link', () => {
  render(<App />);
  const linkElement = screen.getByText(/Konjo/i);
  expect(linkElement).toBeInTheDocument();
});

test("image renders", () => {
  render(<App />);
  const image = screen.getByAltText(/logo/i);
  expect(image).toBeInTheDocument();
})

it('render header', () => {
  // Arrange
  const color = 'purple'

  // Act
  const { container } = render(<Header color={color}></Header>)

  // Assert
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass(`header-${color}`)
})