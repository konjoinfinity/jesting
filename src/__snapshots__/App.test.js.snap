// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render header 1`] = `
<div>
  <header
    class="header header-purple"
  />
</div>
`;
